### DISCUSSION TITLE
Customize/Disable logging in Istio using the Sidecar pattern

### DISCUSSION THREAD

**Discussion by TejasD10** [istio/istio#49362] [Q&A]
### Is this the right place to submit this?

- [X] This is not a security vulnerability or a crashing bug
- [X] This is not a question about how to use Istio

### Bug Description

We observed that Istio (with its default logging configuration) logs the complete URL which, in our case, consists of sensitive information which we wouldn't like to be logged.

Upon research, found that, Istio OOB does not have a way to configure it to not log the queryParameters.

Tried the following but to no avail:

- Set the Istio Log level to "off"

`sidecar.istio.io/logLevel: "off"`

- Set the componentLogLevel to "error" and "off"

`sidecar.istio.io/componentLogLevel curl -s -X POST http://localhost:15000/logging?level=off
`

- Set the agentLogLevel to none.

`sidecar.istio.io/agentLogLevel: "default:none"
`
Log snapshot

```
2024-02-12T23:25:51.216339Z info    cache   returned workload trust anchor from cache   ttl=23h59m59.783666749s
2024-02-12T23:25:51.216647Z info    ads SDS: PUSH request for node:istio-test-1.0.1-7cc4c4f575-ddd12.istio-ns resources:1 size:4.0kB resource:default
2024-02-12T23:25:51.216758Z info    ads SDS: PUSH request for node:istio-test-1.0.1-7cc4c4f575-ddd12.istio-ns resources:1 size:1.1kB resource:ROOTCA
[2024-02-12T23:27:13.262Z] "GET /istio-test?queryParam=Sensitive HTTP/1.1" 403 - rbac_access_denied_matched_policy[none] - "-" 0 19 0 - "10.0.0.0"...
```
Please ignore the RBAC, we have a policy defined that is working as expected, just need guidance with the logging piece.
Help appreciated!

### Version

```Text
GKE version

1.25.15-gke.1115000
```

### Additional Information

The relevant information is provided in the description section.

**Comment by keithmattix** [comment]
You can set your own access log format: https://istio.io/latest/docs/tasks/observability/logs/access-log/#default-access-log-format

**Comment by howardjohn** [comment]
Or just turn off access logs. They are not actually enabled by default either.

**Reply by altdelnow** [reply]
We have a typical issue; the istio skips logging when downstream reports "connection time out issue". We are not able to confirm if istio is even receiving http request. Is there any parameter which can confirm if istio has received http request ?

**Comment by TejasD10** [comment]
I'd like to do this at a sidecar level, as our team does not administer Istio, tried the below to set the accessLogFormat to minimal but the behavior persists. Any other suggestions?
`proxy.istio.io/config: '{ "accessLogFormat": "%RESPONSE_CODE%" }'`

**Comment by howardjohn** [comment]
https://istio.io/latest/docs/reference/config/istio.mesh.v1alpha1/ - only things under ProxyConfig (https://istio.io/latest/docs/reference/config/istio.mesh.v1alpha1/#ProxyConfig) are set in proxy.istio.io. The rest must be set in the mesh config. Generally the `istio` configmap

**Comment by TejasD10** [comment]
Is there a way I can use the config at the mesh level but isolate it to specific workloads? We are on a shared cluster and wouldn't want to affect the other tenants.

**Comment by howardjohn** [comment]
Ah yes, https://istio.io/latest/docs/tasks/observability/telemetry/ can do that. accessLogfile is the simpler/older approach

**Comment by krenaut1** [comment]
If you are feeling brave, you can turn off logging with an EnvoyFilter
```
apiVersion: networking.istio.io/v1alpha3
kind: EnvoyFilter
metadata:
  name: disable-all-access-logs
  namespace: your-workload-namespace
spec:
  workloadSelector:
    labels:
      app: your-workload-label
  configPatches:
    # Disable outbound access logging
    - applyTo: NETWORK_FILTER
      match:
        context: SIDECAR_OUTBOUND
        listener:
          filterChain:
            filter:
              name: "envoy.http_connection_manager"
      patch:
        operation: MERGE
        value:
          typed_config:
            "@type": type.googleapis.com/envoy.config.filter.network.http_connection_manager.v2.HttpConnectionManager
            access_log: []

    # Disable inbound access logging
    - applyTo: NETWORK_FILTER
      match:
        context: SIDECAR_INBOUND
        listener:
          filterChain:
            filter:
              name: "envoy.http_connection_manager"
      patch:
        operation: MERGE
        value:
          typed_config:
            "@type": type.googleapis.com/envoy.config.filter.network.http_connection_manager.v2.HttpConnectionManager
            access_log: []

**Comment by krenaut1** [comment]
Just saw telemetry api answer which is way better answer

```
apiVersion: telemetry.istio.io/v1alpha1
kind: Telemetry
metadata:
  name: your-app-no-log
  namespace: foo
spec:
  selector:
    matchLabels:
      your-app-label: bar
  accessLogging:
  - disabled: true
